2009-02-01  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* sysdeps/mach/hurd/setitimer.c (setitimer_locked): Use common exit path
	instead of returning without unlocking.

---
 sysdeps/mach/hurd/setitimer.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sysdeps/mach/hurd/setitimer.c
+++ b/sysdeps/mach/hurd/setitimer.c
@@ -220,7 +220,7 @@
 	  /* Start up the itimer thread running `timer_thread' (below).  */
 	  if (err = __thread_create (__mach_task_self (),
 				     &_hurd_itimer_thread))
-	    return __hurd_fail (err);
+	    goto out;
 	  _hurd_itimer_thread_stack_base = 0; /* Anywhere.  */
 	  _hurd_itimer_thread_stack_size = __vm_page_size; /* Small stack.  */
 	  if ((err = __mach_setup_thread (__mach_task_self (),
